src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusButton
+
+    enum Size {
+        Small,
+        Large
+    }
+
+    enum Type {
+        Normal,
+        Danger,
+        WithoutBackground,
+        DangerWithoutBackground
+    }
+
+    property bool loading: false
+
+    property int size: StatusButton.Size.Large
+    property int type: StatusButton.Type.Normal
+
+    property alias text: label.text
+    property alias font: label.font
+
+    property real leftPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real rightPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real topPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+    property real bottomPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+
+    signal pressed(var mouse)
+
+
+
+    /// Implementation
+
+    implicitWidth: sensor.width
+    implicitHeight: sensor.height
+
+    radius: 8 // by design
```

Comment:
Same here `// by design` doesn't add any value here

Reply:
The same thing. After 1-2-3 years you could forgot why its `8`?? this annotation for `magic numbers`

Reply:
Alright. I see where you're coming from.

My take would then be: What other reason is there, if not design?

Do we want to put `// by design` comments on every `topMargin`, `padding` etc as well? Because all those things are by design

Reply:
i always do it, but if you want, i can remove it :) its just some info. I think some comments are not superfluous.

Reply:
We can leave them, I think I just need some help from you then to decide in the future, which QML properties receive this comment (since everything is by design).